Automattic/jetpack
# 3rd Party: Add ability for plugins to enable a 'private site' mode

Jetpack doesn't offer a "private site" feature. This PR does not change that.

This adds an easy way for plugins that do offer that feature to disable various features of Jetpack that could "leak" information via the WP.com API, Reader, and Firehose features.

#### Changes proposed in this Pull Request:
* Adds a filter to enable a "private site mode"
* When that filter is enabled, it filters the active modules and available modules to remove the offending modules (without changing the option in the db, so disabling the filter will return the site to the pre-filtered state) and filters the values of sync to the proper value.
* To be done: When the filter is activated, it doesn't immediately sync those values. Need to add a triggering event.

#### Jetpack product discussion
p3btAN-1fd-p2

#### Does this pull request change what data or activity we track or use?
n/a

#### Testing instructions:
* Ensure the JSON API module is enabled.
* Add the filter `add_filter( 'jetpack_private_site_mode', '__return_true');`
* Perform a full sync.
* Try to access the site via public-api, e.g. https://public-api.wordpress.com/rest/v1/sites/[blog-id]/posts/ .
* Expected: See an error that API calls are disabled for this site.

#### Proposed changelog entry for your changes:
* 3rd Party Compatibility: adds a method for "Private Site" plugins to deactivate aspects of Jetpack that make content accessible.
